dask/dataframe/io/parquet/arrow.py

```diff
@@ -582,14 +553,32 @@ def initialize_write(
         if append and division_info is None:
             ignore_divisions = True
 
-        # Extract metadata and get file offset if appending
-        fmd, i_offset, append = cls._get_dataset_offset(
-            path, fs, append, ignore_divisions
-        )
-
-        # Inspect the intial metadata if appending
+        full_metadata = None  # metadata for the full dataset, from _metadata
+        tail_metadata = None  # metadata for at least the last file in the dataset
+        i_offset = 0
+        metadata_file_exists = False
         if append:
-            arrow_schema = fmd.schema.to_arrow_schema()
+            # Extract metadata and get file offset if appending
+            ds = pa_ds.dataset(path, filesystem=fs, format="parquet")
+            i_offset = len(ds.files)
+            if i_offset > 0:
+                try:
+                    with fs.open(fs.sep.join([path, "_metadata"]), mode="rb") as fil:
+                        full_metadata = pq.read_metadata(fil)
+                    tail_metadata = full_metadata
+                    metadata_file_exists = True
+                except OSError:
+                    try:
+                        with fs.open(sorted(ds.files)[-1], mode="rb") as fil:
```

Comment:
```suggestion
                        with fs.open(sorted(ds.files, key=natural_sort_key)[-1], mode="rb") as fil:
```
I think you will need to use `natural_sort_key` to get "proper" ordering here, since we don't zero-pad the counter in the file names. (Sorry if linting is off in my suggestion)